hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/FileIndex.java

```diff
@@ -66,15 +66,19 @@ public class FileIndex {
   private final RowType rowType;
   private final boolean tableExists;
   private final HoodieMetadataConfig metadataConfig;
+  private final HoodieFlinkEngineContext hoodieFlinkEngineContext;
   private final PartitionPruners.PartitionPruner partitionPruner; // for partition pruning
   private final DataPruner dataPruner;                            // for data skipping
   private final int dataBucket;                                   // for bucket pruning
   private List<String> partitionPaths;                            // cache of partition paths
 
   private FileIndex(Path path, Configuration conf, RowType rowType, DataPruner dataPruner, PartitionPruners.PartitionPruner partitionPruner, int dataBucket) {
+    org.apache.hadoop.conf.Configuration hadoopConf = HadoopConfigurations.getHadoopConf(conf);
+
```

Comment:
Can we keep the hadoop conf as member instead?

Reply:
yeah, I will keep the hadoop conf as member

Reply:
I got a warning. Can we ignore it? Or do I need to make any other changes
<img width="1378" alt="image" src="https://user-images.githubusercontent.com/46181516/236094572-fe2e3206-fea6-4304-b06b-8dfae6cc0366.png">

Reply:
Generating the `HoodieFlinkEngineContext` on the fly should be fine.

Reply:
you are right.